cmd/uhc/patch/cmd.go

```diff
@@ -86,124 +54,15 @@ func run(cmd *cobra.Command, argv []string) {
-	cfg, err := config.Load()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Can't load config file: %v\n", err)
-		os.Exit(1)
-	}
-	if cfg == nil {
-		fmt.Fprintf(os.Stderr, "Not logged in, run the 'login' command\n")
-		os.Exit(1)
-	}
-
-	// Check that the configuration has credentials or tokens that don't have expired:
-	armed, err := config.Armed(cfg)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Can't check if tokens have expired: %v\n", err)
-		os.Exit(1)
-	}
-	if !armed {
-		fmt.Fprintf(os.Stderr, "Tokens have expired, run the 'login' command\n")
-		os.Exit(1)
-	}
-
-	// Create the connection:
-	logger, err := util.NewLogger(args.debug)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Can't create logger: %v\n", err)
-		os.Exit(1)
-	}
-	connection, err := client.NewConnectionBuilder().
-		Logger(logger).
-		TokenURL(cfg.TokenURL).
-		Client(cfg.ClientID, cfg.ClientSecret).
-		Scopes(cfg.Scopes...).
-		URL(cfg.URL).
-		User(cfg.User, cfg.Password).
-		Tokens(cfg.AccessToken, cfg.RefreshToken).
-		Insecure(cfg.Insecure).
-		Build()
+	connection, err := util.NewConnection(args.Debug)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Can't create connection: %v\n", err)
 		os.Exit(1)
 	}
 
 	// Create and populate the request:
 	request := connection.Patch().Path(path)
-	for _, parameter := range args.parameter {
-		var name string
-		var value string
-		position := strings.Index(parameter, "=")
-		if position != -1 {
-			name = parameter[:position]
-			value = parameter[position+1:]
-		} else {
-			name = parameter
-			value = ""
-		}
-		request.Parameter(name, value)
-	}
-	for _, header := range args.header {
-		var name string
-		var value string
-		position := strings.Index(header, "=")
-		if position != -1 {
-			name = header[:position]
-			value = header[position+1:]
-		} else {
-			name = header
-			value = ""
-		}
-		request.Header(name, value)
-	}
-	var body []byte
-	if args.body != "" {
-		body, err = ioutil.ReadFile(args.body)
-	} else {
-		body, err = ioutil.ReadAll(os.Stdin)
-	}
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Can't read body: %v\n", err)
-		os.Exit(1)
-	}
-	request.Bytes(body)
-
-	// Send the request:
-	response, err := request.Send()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Can't send request: %v\n", err)
-		os.Exit(1)
-	}
-	status := response.Status()
-	body = response.Bytes()
-	if status < 400 {
-		err = dump.Pretty(os.Stdout, body)
-	} else {
-		err = dump.Pretty(os.Stderr, body)
-	}
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Can't print body: %v\n", err)
-		os.Exit(1)
-	}
-
-	// Save the configuration:
-	cfg.AccessToken, cfg.RefreshToken, err = connection.Tokens()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Can't get tokens: %v\n", err)
-		os.Exit(1)
-	}
-	err = config.Save(cfg)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Can't save config file: %v\n", err)
-		os.Exit(1)
-	}
-
-	// Bye:
-	if status < 400 {
-		os.Exit(0)
-	} else {
-		os.Exit(1)
-	}
+	util.AddParamsAndHeaders(request, args.Parameter, args.Header)
```

Comment:
As I wrote before the "and" in the method name makes it a bit strange. I think it is more natural if we have two methods. It would look like this:

```go
util.AddParameters(request, args.Parameter)
util.AddHeaders(request, args.Header)
util.AddBody(request, args.Body)
```

On more line, but feels more natural, and makes it possible to add only parameters or only headers, should that be needed in some other part of the code.